File: server/models/entity/Booking.go
package entity

import (
	"server/models/dto"
	"time"
)

type Booking struct {
	ID int `json:"id" gorm:"primaryKey"`
	dto.BookingDto
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	EventId   *uint     `json:"-"`
	Event     Event     `gorm:"foreignKey:EventId"`
	VenueId   *uint     `json:"-"`
	Venue     Venue     `gorm:"foreignKey:VenueId"`
}

type BookingRequestDto struct {
	dto.BookingDto
	EventId *uint
	VenueId *uint
}

type BookingResponseDto struct {
	ID int `json:"id" gorm:"primaryKey"`
}

File: client/src/models/dto/Booking.ts
import type { EventResponseDto } from '@/models/dto/Event'
import type { ApiModelDto } from '@/models/Api'
import type { VenueResponseDto } from '@/models/dto/Venue'

export interface BookingResponseDto extends ApiModelDto {
  price: number
  status: string
  Event: EventResponseDto
  Venue: VenueResponseDto
}

export interface BookingRequestDto extends Omit<BookingResponseDto, 'event' | 'venue'> {
  eventId: number
  venueId: number
}

export interface BookingTableItem extends Pick<BookingResponseDto, 'id' | 'price' | 'status'> {
  eventName: EventResponseDto['name']
  eventDescription: EventResponseDto['description']
  venueName: VenueResponseDto['name']
}
